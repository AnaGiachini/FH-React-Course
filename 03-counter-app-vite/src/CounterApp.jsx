import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);
  
  const handleDes = () => setCounter(counter - 1);

  const handleRest = () => setCounter(value);
  

  return (
    <div>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button aria-label='add' onClick={handleAdd}>+1</button>
      <button aria-label='des' onClick={handleDes}>-1</button>
      <button aria-label='reset' onClick={handleRest}>Reset</button>
    </div>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number
}