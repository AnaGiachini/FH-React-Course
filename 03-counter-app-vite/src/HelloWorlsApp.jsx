
export const HelloWorlsApp = () => {
  return (
    <h1>Hello Worls App</h1>
  )
}
