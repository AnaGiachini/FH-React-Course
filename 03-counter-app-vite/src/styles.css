html,
body {
  background-color: #21232a;
  color: white;
  font-size: 1.3rem;
  padding: 70px;
  font-family: Helvetica, Arial, sans-serif;
}

button {
  font-size: 1.3rem;
  padding: 15px;
  margin-right: 5px;
}