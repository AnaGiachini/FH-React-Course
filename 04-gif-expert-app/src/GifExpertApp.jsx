import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

  //Hook para manejar el estado de las categorias
  const [categories, setCategories] = useState(['']);

  //Agrega Categorias
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([...categories, newCategory]);
  }

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />
      
      {
        categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))
      }
    </>
  )
}
