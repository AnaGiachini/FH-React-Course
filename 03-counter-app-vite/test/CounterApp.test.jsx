import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('pruebas sobre CounterApp', () => { 
  
  const initialValue = 10;
  test('debe hacer match con el snapshot', () => {

    const { container } = render(<CounterApp value={1}/>);
    expect(container).toMatchSnapshot();

  });

  test('debe mostrar el valor inicial de 100', () => {
    
    render(<CounterApp value={100}/>);
    expect(screen.getByText(100)).toBeTruthy();

  });

  test('debe incrementar el valor con el boton +1', () => {
    
    render(<CounterApp value={initialValue}/>);
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    expect(screen.getByText(initialValue+1)).toBeTruthy();

  });

  test('debe decrementar el valor con el boton -1', () => {
    
    render(<CounterApp value={initialValue}/>);
    fireEvent.click(screen.getByRole('button', { name: 'des' }));
    // screen.debug();
    expect(screen.getByText(initialValue-1)).toBeTruthy();

  });

  test('debe funcionar el boton reset', () => {
    
    render(<CounterApp value={initialValue}/>);
    fireEvent.click(screen.getByRole('button', { name: 'reset' }));
    expect(screen.getByText(initialValue)).toBeTruthy();

  });
});