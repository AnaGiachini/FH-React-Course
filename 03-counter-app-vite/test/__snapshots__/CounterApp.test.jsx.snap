// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      Counter App
    </h1>
    <h2>
      1
    </h2>
    <button
      aria-label="add"
    >
      +1
    </button>
    <button
      aria-label="des"
    >
      -1
    </button>
    <button
      aria-label="reset"
    >
      Reset
    </button>
  </div>
</div>
`;
